### DISCUSSION TITLE
Way to disable JS8CALL application start and app monitoring?

### DISCUSSION THREAD

**Discussion by msdlogic** [simplyequipped/pyjs8call#3] [General]
This is very handy library 

Is there a method to disable the app launch/monitoring features? at least for Windows? 

In my scenario, I run JS8CALL all the time. It would be running before I start my application using pysj8call and then just connect with TCP.

Also, my application shouldn't kill JS8CALL when it exits. (I did find a way to turn that part off)

I'm glad to dig into the code more. Just wondering if you had a method already.

Really appreciate the library. Very well done.

**Comment by simplyequipped** [comment]
Glad you are finding a use for the library.

Running JS8Call before starting pyjs8call should work just fine since version 0.2.0. This was part of the cross platform testing process when the psutil library was implemented. The appmonitor module will detect that the JS8Call process is already running, and associate with the existing process instead of starting a new process. You should not have to do anything for this to work.

I should be able to add an attribute to the appmonitor module to control whether the JS8Call process is terminated when stopping pyjs8call. I just made the code change in the develop branch, but have not tested it yet. If things go well it should be included in the next version release. This is a use case I had not considered, so thanks for brining it up.

I would be curious to know what you are doing now to prevent the JS8Call process from terminating when stopping pyjs8call. Maybe you figured out a better approach :)

**Comment by msdlogic** [comment]
Thanks for reply. I basically took the same approach.. returning right away from "stop def". So far, with my testing, that prevents JS8CALL from exiting ok.

However, in my scenario (Windows 11), if I run JS8CALL first.. and then run my application using pyjs8call, I get a screen popup warning from JS8CALL that says "Another Instance may be running, try to remove stale lock file". Somewhere, pyjs8call seems to still try to launch JS8CALL, I'm on 0.2.1 

It would be great to be able to turn off starting, monitoring and automatically closing the JS8CALL application. This way, I can have my application report to the user that it cannot connect to JS8CALL over TCP and suggest the user start JS8CALL manually. This is similar to how other apps like JS8spotter work. 

I'll try to troubleshoot some more on this! Thanks!

**Comment by simplyequipped** [comment]
I just tested this on Windows and saw the same issue. Apparently my prior testing wasn't as fool-proof as I assumed. Good catch!

The issue was in the [pyjs8call.appmonitor.AppMonitor._find_running_js8call_process](https://github.com/simplyequipped/pyjs8call/blob/a44daf29da3aa6c773b837ab27f068bdb2f344c6/pyjs8call/appmonitor.py#L294) function. I was only checking for the process name 'js8call' as startup. That works for Linux, but not for Windows which names the process 'js8call.exe'. I pushed a change on the develop branch to check for both process names, and it seems to work for me on Windows now.

Let me know if that fixes the issue for you.

**Comment by msdlogic** [comment]
Yes! Works great! Thanks for fixing that so fast. I haven't done much Windows dev so my head was spinning, but it totally makes sense now that you explain it.

Again.. nice work here. The transmit queueing works great. It's not included by other JS8CALL APIs that I could find.

**Reply by simplyequipped** [reply]
I found it useful to offload things like outgoing message queuing from the application to the API. I'm glad you have found it useful as well.
